gcloud-java-examples/src/main/java/com/google/gcloud/examples/DnsExample.java

```diff
@@ -0,0 +1,516 @@
+import com.google.gcloud.dns.ProjectInfo;
+import com.google.gcloud.dns.Zone;
+import com.google.gcloud.dns.ZoneInfo;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An example of using Google Cloud DNS.
+ *
+ * <p>This example creates, deletes, gets, and lists zones, and creates and deletes DNS records of
+ * type A.
+ *
+ * <p>Steps needed for running the example:<ol>
+ * <li>login using gcloud SDK - {@code gcloud auth login}.</li>
+ * <li>compile using maven - {@code mvn compile}</li>
+ * <li>run using maven - {@code mvn exec:java
+ * -Dexec.mainClass="com.google.gcloud.examples.dns.DnsExample"
+ * -Dexec.args="[<project_id>]
+ * create <zone_name> <dns_name> <description> |
+ * get <zone_name> |
+ * delete <zone_name> |
+ * list [<zone_name> [changes [descending | ascending] | records]] |
+ * add-record <zone_name> <record_name> <ip> <ttl> |
+ * delete-record <zone_name> <record_name> <ip> [<ttl>] |
+ * quota</li>
+ * </ol>
+ *
+ * <p>The first parameter is an optional {@code project_id} (logged-in project will be used if not
+ * supplied). Second parameter is a DNS operation (list, delete, create,...) and can be used to
+ * demonstrate the usage. The remaining arguments are specific to the operation. See each action's
+ * run method for the specific interaction.
+ */
+public class DnsExample {
+
+  private static final Map<String, DnsAction> ACTIONS = new HashMap<>();
+
+  private interface DnsAction {
+    void run(Dns dns, String... args);
+
+    String params();
+
+    boolean check(String... args);
+  }
+
+  private static class CreateZoneAction implements DnsAction {
+
+    /**
+     * Creates a zone with the provided name, dns name and description (in this order).
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String dnsName = args[1];
+      String description = args[2];
+      ZoneInfo zoneInfo = ZoneInfo.builder(zoneName)
+          .dnsName(dnsName)
+          .description(description)
+          .build();
+      Zone zone = dns.create(zoneInfo);
+      System.out.printf("Successfully created zone with name %s which was assigned ID %s.%n",
+          zone.name(), zone.id());
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name> <dns_name> <description>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 3;
+    }
+  }
+
+  private static class ListZonesAction implements DnsAction {
+
+    /**
+     * Lists all zones within the project.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      Iterator<Zone> zoneIterator = dns.listZones().iterateAll();
+      if (zoneIterator.hasNext()) {
+        System.out.println("The project contains the following zones:");
+        System.out.println("Name\tID\tDNS Name\tCreated\tDesription");
```

Comment:
Tabbing could lead to less than ideal results if fields vary in size significantly.  Perhaps each list item could be in some variation of the following format:
Name: <>, ID: <>, DNS name: <>
    Created: <>
    Description: <>